### DISCUSSION TITLE
ubuntu python issues

### DISCUSSION THREAD

**Discussion by Insistentelk** [vladmandic/sdnext#3410] [Q&A]
Any one figure out a way to force automatic to detect a correct version of python. If Ubuntu auto updated you to the 24.04 version with python 3.12.3 as default. Even if you mange to install the and compatible version of python and set it as the version the code still keep garbing python version 3.12.3.

Has any one figured out how to get this to work on Ubuntu 24.04 or is  a full reinstall on a older version that doesn't have python 3.12.3 the only option.

Edit: Apparently on 24.04 you need to use for PyNEV

curl https://pyenv.run | bash
sudo apt-get install libssl-dev
sudo apt-get install liblzma-dev
sudo apt-get install libffi-dev
sudo apt-get install ncurses-dev

as apparently their are dependencies not properly located and installed.

**Comment by brknsoul** [comment]
if you have python3.10 or python3.11 installed (check by `/usr/bin/python3.10 --version` or `/usr/bin/python3.11 --version`)
then simply `export PYTHON=/usr/bin/python3.10` (or `.11`) before running webui.sh.

If not, then `sudo apt install python3.10`.

**Comment by vladmandic** [comment]
or use `pyenv` - i use it all the time to switch between multiple python versions.
